internal/cmd/server/common.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2024 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+compliance with the License. You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is
+distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing permissions and limitations under the
+License.
+*/
+
+package server
+
+import (
+	"context"
+	"errors"
+	"log/slog"
+	"os"
+	"strings"
+)
+
+func processBackendToken(
```

Comment:
If we are going to extract this logic to a common place I'd suggest we do it for both adding the flags and getting them, something similar to what we do for the [network listening flags](https://github.com/openshift-kni/oran-o2ims/blob/main/internal/network/flags.go). For example, we could have a function to add the flags to a flag set:

```go
// AddTokenFlags adds to the given flagset the flags needed to configure a token ...
func AddTokenFlags(set *pflag.FlagSet, name string) {
         ...
}
```

And then a function to get the value of the token from a flag set:

```go
// GetTokenFlag gets the value of the a token flag ...
func GetTokenFlag(set *pflag.FlagSet, name string) (result string, err error) {
        ...
}
```

That way if we ever need to change how we add or get these flags it will all be in the same place.

Reply:
Updated. @jhernand , kindly confirm this is what you had in mind.

Reply:
Looks good. I think it can be even better if we pass the `pflag.FlagSet` object to the `GetTokenFlag` function because then that function can also extract the values of the flags from that flag set.